base_report_to_printer/models/printing_server.py

```diff
@@ -0,0 +1,232 @@
+    port = fields.Integer(
+        default=631, required=True, help='Port of the server.')
+    active = fields.Boolean(
+        default=True, help='If checked, this server is useable.')
+    printer_ids = fields.One2many(
+        comodel_name='printing.printer', inverse_name='server_id',
+        string='Printers List',
+        help='List of printers available on this server.')
+
+    @api.multi
+    def _open_connection(self, raise_on_error=False):
+        self.ensure_one()
+        connection = False
+        try:
+            connection = cups.Connection(host=self.address, port=self.port)
+        except:
+            message = _("Failed to connect to the CUPS server on %s:%s. "
+                        "Check that the CUPS server is running and that "
+                        "you can reach it from the Odoo server.") % (
+                            self.address, self.port)
+            _logger.warning(message)
+            if raise_on_error:
+                raise exceptions.UserError(message)
+
+        return connection
+
+    @api.multi
+    def action_update_printers(self):
+        return self.update_printers()
+
+    @api.multi
+    def update_printers(self, domain=None, raise_on_error=False):
+        mapping = {
+            3: 'available',
+            4: 'printing',
+            5: 'error'
+        }
+
+        if domain is None:
+            domain = []
+
+        servers = self
+        if not self:
+            servers = self.search(domain)
+
+        res = True
+        for server in servers:
+            connection = server._open_connection(raise_on_error=raise_on_error)
+            if not connection:
+                server.printer_ids.write({'status': 'server-error'})
+                res = False
+                continue
+
+            # Update Printers
+            printers = connection.getPrinters()
+            existing_printers = dict([
+                (printer.system_name, printer)
+                for printer in server.printer_ids
+            ])
+            updated_printers = []
+            for name, printer_info in printers.iteritems():
+                printer_values = {
+                    'name': printer_info['printer-info'],
+                    'system_name': name,
+                    'server_id': server.id,
+                    'uri': printer_info.get('device-uri', ''),
+                    'status': mapping.get(printer_info.get('printer-state'),
+                                          'unknown'),
+                    'status_message': printer_info.get(
+                        'printer-state-message', ''),
+                    'model': printer_info.get('printer-make-and-model', False),
+                    'location': printer_info.get('printer-location', False),
+                }
+                updated_printers.append(name)
+                if name not in existing_printers.keys():
+                    self.env['printing.printer'].sudo().create(printer_values)
```

Comment:
I used sudo here because on our module, we run `update_jobs` (which runs `update_printers`) juste after sending a job to the printer, and not all users have the right to create a printer.

In this PR, I simply replaced the `update_printers` cron by `update_jobs`, which is run every minute, so I think the sudo is not needed anymore (unless we re-add the `update_jobs` call after each print, which doesn't seem useful).